lib/src/crawl_async.dart

```diff
@@ -34,7 +34,7 @@ final _empty = Future<Null>.value(null);
 /// have not completed. If the [edges] callback needs to be limited or throttled
 /// that must be done by wrapping it before calling [crawlAsync].
 Stream<V> crawlAsync<K, V>(Iterable<K> roots, FutureOr<V> Function(K) readNode,
-    FutureOr<Iterable<K>> Function(K, V) edges) {
+    FutureOr<Iterable<K>?> Function(K, V) edges) {
```

Comment:
@natebosch I am a bit surprised we allow returning null here? should we retain that behavior?

Reply:
No, I think we should restrict it - we previously allowed `null` and `[]` to mean the same thing, and there is no need to do that in a null safe world.

I'm on the fence about statically typing it as non-nullable, but dynamically allowing null to maintain backwards compatibility for opt-out code. Even with a major version bump it might be friendly to do so.